v3.1.0: release notes (#1767)
AviatrixSystems/terraform-provider-aviatrix: CHANGELOG.md

## 3.1.0 (Unreleased)
### Notes:
- Supported Controller version: **UserConnect-7.1**
- Supported Terraform version: **v1.x**

### Features:
#### Gateway:

> 1. Implemented support of configuring GRO/GSO in **aviatrix_gateway**

Reviewer: ```suggestion
1. Implemented support for configuring GRO/GSO in **aviatrix_gateway**
```

#### Multi-Cloud Transit:

> 1. Implemented support of configuring Local Identifier in **aviatrix_site2cloud**, **aviatrix_spoke_external_device_conn** and **aviatrix_transit_external_device_conn**
> 2. Implemented support of configuring GRO/GSO in **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**
> 3. Implemented support of BGP over GRE on a BGP Spoke in **aviatrix_spoke_external_device_conn**

Reviewer: ```suggestion
1. Implemented support for configuring Local Identifier in **aviatrix_site2cloud**, **aviatrix_spoke_external_device_conn** and **aviatrix_transit_external_device_conn**
2. Implemented support for configuring GRO/GSO in **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**
3. Implemented support for BGP over GRE on a BGP Spoke in **aviatrix_spoke_external_device_conn**
```

   - **aviatrix_edge_neo**
   - **aviatrix_edge_neo_ha**
6. Added a new attribute in **aviatrix_spoke_gateway** to support GCP spoke global VPC:
   - ``enable_global_vpc``
7. Implemented new resources to support GCP global VPC excluded instance and tagging settings:
   - **aviatrix_global_vpc_excluded_instance**
   - **aviatrix_global_vpc_tagging_settings**
8. Added a new attribute in **aviatrix_edge_spoke_external_device_conn** to support BGP over WAN underlay:
   - ``enable_edge_underlay``
9. Added a new attribute in **aviatrix_edge_spoke_external_device_conn** to support Prepend AS Path:
   - ``prepend_as_path``
10. Added a new attribute in **aviatrix_edge_spoke_transit_attachment** to support multiple WAN interfaces:
    - ``edge_wan_interfaces``
11. Implemented a new data source to get Edge gateway WAN interface IP address:
    - **aviatrix_edge_gateway_wan_interface_discovery**

#### Settings:
1. Implemented a new data source to collect controller metadata:
   - **aviatrix_controller_metadata**

#### Copilot:
1. Implemented new resources to support QoS class and QoS policy list:
   - **aviatrix_qos_class**
   - **aviatrix_qos_policy_list**

### Enhancements:

> 1. Added support of ``fqdn`` as one of the attributes under ``selector`` in **aviatrix_smart_group**
> 2. Added support of "#" for Azure gateway tags
> 3. Added support of enabling BGP over LAN for Azure transit in update in **aviatrix_transit_gateway**

Reviewer: ```suggestion
1. Added support for ``fqdn`` as one of the attributes under ``selector`` in **aviatrix_smart_group**
2. Added support for the "#" character in Azure gateway tags
3. Added support for enabling BGP over LAN for Azure transit in update in **aviatrix_transit_gateway**
```

4. Changed ``cloud_type``, ``account_name``, ``insane_mode`` and ``insane_mode_az`` to "ForceNew" in **aviatrix_gateway**, **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**

> 5. Removed default value of ``bgp_lan_interfaces_count`` in **aviatrix_transit_gateway**

Reviewer: ```suggestion
5. Removed ``bgp_lan_interfaces_count``'s default value in **aviatrix_transit_gateway**
```

> 6. Remove the option to config proxies in private mode config in **aviatrix_controller_private_mode_config**

Reviewer: ```suggestion
6. Removed the option to config proxies in private mode config in **aviatrix_controller_private_mode_config**
```

> 7. Restored following attributes in **aviatrix_cloudn_transit_gateway_attachment**:

Reviewer: ```suggestion
7. Restored the following attributes in **aviatrix_cloudn_transit_gateway_attachment**:
```

> 6. Remove option to config proxies in private mode config in **aviatrix_controller_private_mode_config**
> 7. Restored following attributes in **aviatrix_cloudn_transit_gateway_attachment**:

Reviewer: ```suggestion
6. Removed the option to config proxies in private mode config in **aviatrix_controller_private_mode_config**
7. Restored the following attributes in **aviatrix_cloudn_transit_gateway_attachment**:
```

> 6. Remove option to config proxies in private mode config in **aviatrix_controller_private_mode_config**
> 7. Restored following attributes in **aviatrix_cloudn_transit_gateway_attachment**:

Reviewer: ```suggestion
6. Removed the option to config proxies in private mode config in **aviatrix_controller_private_mode_config**
7. Restored the following attributes in **aviatrix_cloudn_transit_gateway_attachment**:
```

   - ``enable_dead_peer_detection``
   - ``enable_learned_cidrs_approval``
   - ``approved_cidrs``

> 8. Added support of "ddog-gov.com" in **aviatrix_datadog_agent**
> 9. Added support of connection with HA in **aviatrix_edge_spoke_external_device_conn**

Reviewer: ```suggestion
8. Added support for "ddog-gov.com" in **aviatrix_datadog_agent**
9. Added support for connection with HA in **aviatrix_edge_spoke_external_device_conn**
```
